### PR TITLE
Add team icon

### PR DESCRIPTION
**Pull request by xsyann**
Add a team icon next to the username and in the user list:

Screenshot 2022-09-09 at 11 44 41

Screenshot 2022-09-09 at 11 44 53

Here is an example of how to create a guild:

1. Upload a team icon as an asset in the editor
2. Get the icon id in the `Files` collection
3. Create the guild in the `guilds` collection (replace the `icon.fileId` with the correct id)

```json
{
  "_id": "gui_bZdBbTt5xRN7nAup5",
  "createdAt": {
    "$date": {
      "$numberLong": "1663667575055"
    }
  },
  "name": "Refty",
  "description": "Le recrutement nouvelle génération.",
  "website": "https://www.refty.co/",
  "logo": "https://d3m90ikr40n4n5.cloudfront.net/refty/refty-white.png",
  "icon": {
    "fileId": "hWssnvrc5KexTbpvt",
    "width": 16,
    "height": 16
  }
}
```

4. Add the attribute `guildId: "gui_bZdBbTt5xRN7nAup5"` to a `user`

### CODE REVIEW

File: core/client/components/character-name-text.js [hunk 1]
```diff
@@ -64,8 +56,43 @@ class CharacterNameText extends Phaser.GameObjects.Container {
...
+  }
+
+  updateIconPosition() {
+    if (this.icon) this.icon.setX(-(this.name.width + this.icon.displayWidth) / 2 - iconMargin);
+  }
+
+  setIcon(icon) {
+    this.destroyIcon();
+    if (icon) {
+      game.scene.getScene('BootScene').loadImagesAtRuntime([icon], () => {
```
Comment (xsyann) [inline core/client/components/character-name-text.js:70]: @Donorhan Not sure about this but I think loading all guilds icons at preload is not a good solution either (can be better but depending on the number of existing guilds).
I thought about using this solution (loading at runtime) and optimize it a bit with a local cache but it seems that the LoaderPlugin already has a cache system.
Do you have a suggestion on what would be the optimal solution in your idea?

Comment (xsyann) [reply core/client/components/character-name-text.js:70]: @Donorhan I added an example of a guild with an icon to insert in DB if you want to test more easily this PR ;)

Comment (Donorhan) [reply core/client/components/character-name-text.js:70]: Thank you! As soon as I find some time I'll take care of it 😬

Comment (ramnes) [reply core/client/components/character-name-text.js:70]: FWIW this is the most blocking of our PRs, we have a few branches that depend on it (such as #141 but also others that we didn't PR yet). 😅

File: core/server/users.js [hunk 2]
```diff
@@ -18,7 +18,11 @@ Meteor.publish('users', function (levelId) {
     };
   }
 
-  return Meteor.users.find(filters, { fields: mainFields });
+  const users = Meteor.users.find(filters, { fields: mainFields });
+  const guildIds = users.map(u => u.guildId).filter(Boolean);
+  const guilds = Guilds.find({ _id: { $in: [...new Set(guildIds)] } });
+
+  return [users, guilds];
```
Comment (Donorhan) [inline core/server/users.js:25]: I was wondering, is this ok in terms of performance? I haven't tested amplitude so maybe you have some info about it

**Comment by Donorhan** [comment]
Thanks, it's perfect!

(And sorry for the delay 😬)

**Pull request merged**